diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b27e3c..78d3952 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -168,9 +168,8 @@ message(STATUS "*** Looking for external libraries")
 ## Find LAPACK and BLAS libraries, or their optimised versions
 ##
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake_aux/Modules/")
 set(COOT_HAS_BLAS false)
 set(COOT_HAS_LAPACK false)
 
 if(APPLE)
   message(STATUS "Detected macOS")
@@ -255,10 +254,15 @@ else()
     set(FlexiBLAS_FOUND false)
   endif()

+  if(0)
   include(COOT_FindMKL)
   include(COOT_FindOpenBLAS)
   include(COOT_FindBLAS)
   include(COOT_FindLAPACK)
+  else()
+    find_package(BLAS REQUIRED)
+    find_package(LAPACK REQUIRED)
+  endif()

   message(STATUS "FlexiBLAS_FOUND = ${FlexiBLAS_FOUND}" )
   message(STATUS "      MKL_FOUND = ${MKL_FOUND}"       )
@@ -379,7 +383,8 @@ set(COOT_LIBS ${COOT_LIBS} ${OpenCL_LIBRARIES})
     set(COOT_INCLUDE_DIRS ${COOT_INCLUDE_DIRS} ${OpenCL_INCLUDE_DIRS})

     # Now try to find clBLAS.
-    include(COOT_FindCLBLAS)
+    find_package(clBLAS REQUIRED)
+    set(CLBLAS_FOUND "YES")
     if(CLBLAS_FOUND)
       set(COOT_LIBS ${COOT_LIBS} ${CLBLAS_LIBRARIES})
       set(COOT_INCLUDE_DIRS ${COOT_INCLUDE_DIRS} ${CLBLAS_INCLUDE_DIRS})


